compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/semantics/model/types/BUnionType.java

```diff
@@ -221,4 +221,11 @@ public boolean isPureType() {
                         ((BUnionType) type).memberTypes.stream() : Stream.of(type))
                 .collect(Collectors.toCollection(LinkedHashSet::new));
     }
+
+    public BType normalize() {
```

Comment:
Should we add this method here? By definition, a union type has > 1 member types, IMO this isn't part of the union type?

Reply:
If we are to go by that definition we need to create a union builder. Since it's a too much of a change, and since we have to normalize most of the union instances I think it's a responsibility of the union class.

Reply:
IMO, we shouldn't even be creating a union type if we have only one type. What we usually do when creating a union type IINM is creating a set and then adding members to it, then checking the number of members and creating a union type if > 1, else using the only member.

We create a union type in this scenario and in #20180 just because it makes it easier to check assignability right? IMO, we don't need a union builder, but can just do this in line since it won't be that common?

Reply:
The builder is not to support this use case but to support the common case. What build would do it what you said in the first paragraph, but we don't have to manually create sets and stuff, and it will also stop people like me creating probable single-member unions.

I originally created the union not to check assignability, but because I wasn't aware of the optimization we seemingly suppose to do for unions. 

OK will inline this.